vscode/microsoft-kiota/src/util.ts

```diff
@@ -92,4 +95,118 @@ export async function handleMigration(
           vscode.window.showErrorMessage(vscode.l10n.t(`Migration failed: ${error}`));
       }
   });
-}
\ No newline at end of file
+}
+
+export function getSanitizedString(rawValue?: string): string| undefined{
+  return rawValue?.replace(/[^a-zA-Z0-9_]+/g, '');
+};
+
+export function parseGenerationType(generationType: string | QuickPickItem | undefined): GenerationType {
+    if(typeof generationType !== 'string') {
+        throw new Error('generationType has not been selected yet');
+    }
+    switch(generationType) {
+        case "client":
+            return GenerationType.Client;
+        case "plugin":
+            return GenerationType.Plugin;
+        case "apimanifest":
+            return GenerationType.ApiManifest;
+        default:
+            throw new Error(`Unknown generation type ${generationType}`);
+    }
+}
+
+export function parseGenerationLanguage(value: string): KiotaGenerationLanguage {
+    switch (value.toLowerCase()) {
+      case "csharp":
+          return KiotaGenerationLanguage.CSharp;
+      case "java":
+          return KiotaGenerationLanguage.Java;
+      case "typescript":
+          return KiotaGenerationLanguage.TypeScript;
+      case "php":
+          return KiotaGenerationLanguage.PHP;
+      case "python":
+          return KiotaGenerationLanguage.Python;
+      case "go":
+          return KiotaGenerationLanguage.Go;
+      case "swift":
+          return KiotaGenerationLanguage.Swift;
+      case "ruby":
+          return KiotaGenerationLanguage.Ruby;
+      case "cli":
+          return KiotaGenerationLanguage.CLI;
+      default:
+          throw new Error("unknown language");
+    }
+}
+
+export function parsePluginType(values: string[]): KiotaPluginType[] {
+    return values.map(value => {
+        switch (value.toLowerCase()) {
+            case "openai":
+                return KiotaPluginType.OpenAI;
+            case "apimanifest":
+                return KiotaPluginType.ApiManifest;
+            case "apiplugin":
+                return KiotaPluginType.ApiPlugin;
+            default:
+                throw new Error(`unknown plugin type: ${value}`);
+        }
+    });
+}
+
+export function allGenerationLanguagesToString(): string[] {
+  let allSupportedLanguages: string[] = allGenerationLanguages.map(langEnum => KiotaGenerationLanguage[langEnum]);
+  return allSupportedLanguages;
+}
+
+export function validateDeepLinkQueryParams(queryParameters: Record<string, string>):
+  Record<string, string|undefined> 
+{
+  let validQueryParams: Record<string, string|undefined> = {};
+  const descriptionUrl = queryParameters["descriptionurl"];
+  const name = getSanitizedString(queryParameters["name"]);
+  const source = getSanitizedString(queryParameters["source"]);
+  let lowercasedKind: string = queryParameters["kind"]?.toLowerCase();
+  let validKind: string | undefined = ["plugin", "client"].indexOf(lowercasedKind) > -1 ? lowercasedKind : undefined ;
+  if (!validKind){
```

Comment:
It's intended. If !validkind, just show message to user and continue without skipping predefined questions answered by kind.

Reply:
What would the user do with the information?

Reply:
Yeah, good question. It would be good for user to know that they provided the wrong kind of param and therefore expect the quick picker to re-ask that question, but maybe it will also look like an error and confuse the use on way forward. So maybe the right approach would be to console.log or send to telemetry for debugging purposes. I could use some thoughts here.

Reply:
I have modified this to send the errors as telemetry instead for debugging purposes and not show user any errors.